tests/components/canary/test_init.py

```diff
@@ -1,37 +1,82 @@
 """The tests for the Canary component."""
-from requests import HTTPError
+from requests import ConnectTimeout
 
-from homeassistant.components.canary import DOMAIN
+from homeassistant.components.camera.const import DOMAIN as CAMERA_DOMAIN
+from homeassistant.components.canary.const import CONF_FFMPEG_ARGUMENTS, DOMAIN
+from homeassistant.config_entries import (
+    ENTRY_STATE_LOADED,
+    ENTRY_STATE_NOT_LOADED,
+    ENTRY_STATE_SETUP_RETRY,
+)
+from homeassistant.const import CONF_PASSWORD, CONF_TIMEOUT, CONF_USERNAME
 from homeassistant.setup import async_setup_component
 
-from tests.async_mock import patch
+from . import YAML_CONFIG, init_integration
 
+from tests.async_mock import patch
 
-async def test_setup_with_valid_config(hass, canary) -> None:
-    """Test setup with valid YAML."""
-    await async_setup_component(hass, "persistent_notification", {})
-    config = {DOMAIN: {"username": "test-username", "password": "test-password"}}
 
+async def test_import_from_yaml(hass, canary) -> None:
+    """Test import from YAML."""
     with patch(
-        "homeassistant.components.canary.alarm_control_panel.setup_platform",
+        "homeassistant.components.canary.async_setup_entry",
         return_value=True,
-    ), patch(
-        "homeassistant.components.canary.camera.setup_platform",
-        return_value=True,
-    ), patch(
-        "homeassistant.components.canary.sensor.setup_platform",
+    ):
+        assert await async_setup_component(hass, DOMAIN, {DOMAIN: YAML_CONFIG})
+        await hass.async_block_till_done()
+
+    entries = hass.config_entries.async_entries(DOMAIN)
+    assert len(entries) == 1
+
+    assert entries[0].data[CONF_USERNAME] == "test-username"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6184085)